frontend/src/features/mapSideBar/containers/MapSideBarContainer.tsx

```diff
@@ -196,31 +196,39 @@ const MapSideBarContainer: React.FunctionComponent<IMapSideBarContainerProps> =
     if (latLng) {
       parcelLayerService.findOneWhereContains(latLng).then(resp => {
         const properties = getIn(resp, 'features.0.properties');
-        if ((properties?.PIN || properties?.PID) && isParcel) {
-          const query: any = { pin: properties?.PIN, pid: properties.PID };
-          fetchParcelsDetail(query)(dispatch).then((resp: any) => {
-            const matchingParcel: any = resp?.data?.length ? _.first(resp?.data) : undefined;
-            if (!!nameSpace && !!formikRef?.current?.values && !!matchingParcel?.id) {
-              const { resetForm, values } = formikRef.current;
-              resetForm({ values: setIn(values, nameSpace, matchingParcel) });
-              toast.dark('Found matching parcel within PIMS. Form data will be pre-populated.', {
-                autoClose: 7000,
-              });
-            } else {
-              const response = properties.PID
-                ? parcelLayerService.findByPid(properties.PID)
-                : parcelLayerService.findByPin(properties.PIN);
-
-              handleParcelDataLayerResponse(response, dispatch, latLng);
-            }
-          });
+        if (properties?.PIN || properties?.PID) {
+          if (isParcel) {
+            const query: any = { pin: properties?.PIN, pid: properties.PID };
+            fetchParcelsDetail(query)(dispatch).then((resp: any) => {
+              const matchingParcel: any = resp?.data?.length ? _.first(resp?.data) : undefined;
+              if (!!nameSpace && !!formikRef?.current?.values && !!matchingParcel?.id && isParcel) {
+                const { resetForm, values } = formikRef.current;
+                resetForm({ values: setIn(values, nameSpace, matchingParcel) });
+                toast.dark('Found matching parcel within PIMS. Form data will be pre-populated.', {
+                  autoClose: 7000,
+                });
+              } else {
+                parcelLayerSearch(properties, latLng);
+              }
+            });
+          } else {
+            parcelLayerSearch(properties, latLng);
```

Comment:
even though there is very little that can be populated for buildings, we still want to search by pid/pin (just not within PIMS).